src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs

```diff
@@ -120,12 +104,31 @@ private FieldType DeriveField(GraphFieldInfo fieldInfo)
                 Description = fieldInfo.Description,
                 DeprecationReason = fieldInfo.DeprecationReason,
                 DefaultValue = fieldInfo.DefaultValue,
+                Metadata = DeriveFieldTypeMetaData(fieldInfo.AttributeProvider.GetCustomAttributes(false)),
                 Type = GetType(fieldInfo.Type),
                 Arguments = new QueryArguments(fieldInfo.Arguments.Where(arg => !arg.IsInjected).Select(DeriveArgument)),
                 Resolver = FieldResolverFactory(fieldInfo),
             };
         }
 
+        private IDictionary<string, object> DeriveFieldTypeMetaData(object[] attributes)
```

Comment:
I think we could also write this one like this:

```
if (attributes == null)
   return default;

return attributes
   .OfType<FieldTypeMetaDataAttribute>()
   .ToDictionary(x => x.Key(), x.Value());
```

Maybe we could consider in-lining it in the `DeriveType` that way as well.

Reply:
Very good remark